### PR TITLE
Add data field in FCM notification payload

### PR DESCRIPTION
**Pull request by bavishhh**
Add notification-type (event, blog-entry or news-entry) and event-id (if the notification is sent for an event).
#206 
@pulsejet This is not tested as I don't have access to Firebase account!

### CODE REVIEW

File: other/management/commands/push-notify.py [hunk 1]
```diff
@@ -57,11 +58,19 @@ def handle(self, *args, **options):
 
                 # Get title
                 title = "InstiApp"
+                notification_type = "event"
+                event_id = uuid4()
                 actor = notification.actor
                 if isinstance(actor, Event):
                     title = actor.name
-                if isinstance(actor, BlogEntry) or isinstance(actor, NewsEntry):
+                    event_type = "event"
```
Comment (pulsejet) [inline other/management/commands/push-notify.py]: Should `notification_type`

File: other/management/commands/push-notify.py [hunk 2]
```diff
@@ -57,11 +58,19 @@ def handle(self, *args, **options):
 
                 # Get title
                 title = "InstiApp"
+                notification_type = "event"
```
Comment (pulsejet) [inline other/management/commands/push-notify.py]: Don't assume defaults; the object could be a None, in which case none of the following might be triggered.

File: other/management/commands/push-notify.py [hunk 3]
```diff
@@ -57,11 +58,19 @@ def handle(self, *args, **options):
...
+                event_id = uuid4()
```
Comment (pulsejet) [inline other/management/commands/push-notify.py]: Do not assume defaults, same as above

File: other/management/commands/push-notify.py [hunk 4]
```diff
@@ -57,11 +58,19 @@ def handle(self, *args, **options):
...
                 actor = notification.actor
                 if isinstance(actor, Event):
                     title = actor.name
-                if isinstance(actor, BlogEntry) or isinstance(actor, NewsEntry):
+                    event_type = "event"
+                    event_id = actor.id
```
Comment (pulsejet) [inline other/management/commands/push-notify.py]: Cast this to a string

File: other/management/commands/push-notify.py [hunk 5]
```diff
@@ -57,11 +58,19 @@ def handle(self, *args, **options):
...
+                if isinstance(actor, BlogEntry):
                     title = actor.title
+                    notification_type = "blog-entry"
+                if isinstance(actor, NewsEntry):
+                    title = actor.title
+                    notification_type = "news-entry"
```
Comment (pulsejet) [inline other/management/commands/push-notify.py]: We want the id in this case too (BlogEntry and NewsEntry). Return them as well.

**Review by pulsejet — CHANGES_REQUESTED** [review]
Please make these changes and fix the couple of errors reported by codacy

File: other/management/commands/push-notify.py [hunk 6]
```diff
@@ -98,6 +108,10 @@ def handle(self, *args, **options):
                             "data": {
                                 "primaryKey": 1
                             }
+                        },
+                        "data": {
+                            "type": notification_type,
+                            "id": notification_id
```
Comment (pulsejet) [inline other/management/commands/push-notify.py:114]: Should this be outside, or inside notification? Please check once.

Comment (bavishhh) [reply other/management/commands/push-notify.py:114]: In the [firebase docs](https://firebase.google.com/docs/cloud-messaging/concept-options#notification-messages-with-optional-data-payload) the data is outside the notification.

**Review by unstablebrainiac — APPROVED** [review]
Looks good to me

**Review by pulsejet — APPROVED** [review]

**Pull request merged**